packages/quantic/force-app/main/default/lwc/quanticPager/e2e/pagerObject.ts

```diff
@@ -0,0 +1,72 @@
+import {Page} from '@playwright/test';
+import {isUaCustomEvent} from '../../../../../../playwright/utils/requests';
+
+export class PagerObject {
+  constructor(public page: Page) {
+    this.page = page;
+  }
+
+  get nextPageButton() {
+    return this.page.locator('button[title="Next Page"]');
+  }
+
+  get previousPageButton() {
+    return this.page.locator('button[title="Previous Page"]');
+  }
+
+  get pageButtons() {
+    return this.page.locator('c-quantic-number-button button');
+  }
+
+  pageButton(pageNumber: number) {
+    return this.pageButtons.nth(pageNumber - 1);
+  }
+
+  async goToLastPage() {
+    await this.pageButtons.nth(-1).click();
+  }
+
+  async clickNextPageButton() {
+    await this.nextPageButton.click();
+  }
+
+  async clickPreviousPageButton() {
+    await this.previousPageButton.click();
+  }
+
+  async clickPageNumberButton(pageNumber: number) {
+    await this.pageButtons.nth(pageNumber - 1).click();
+  }
+
+  async waitForPagerUaAnalytics(eventValue) {
+    const uaRequest = this.page.waitForRequest((request) => {
+      if (isUaCustomEvent(request)) {
+        const requestBody = request.postDataJSON?.();
+        const expectedFields = {
+          eventType: 'getMoreResults',
+          eventValue: eventValue,
+        };
+        return Object.keys(expectedFields).every(
+          (key) => requestBody?.[key] === expectedFields[key]
+        );
+      }
+      return false;
+    });
+    return uaRequest;
+  }
```

Comment:
Here it could be nice to specify the type of return too, also you don't have to create the temp variable, was there a reason?

```suggestion
  async waitForPagerUaAnalytics(eventValue): Promise<request> {
    return this.page.waitForRequest((request) => {
      if (isUaCustomEvent(request)) {
        const requestBody = request.postDataJSON?.();
        const expectedFields = {
          eventType: 'getMoreResults',
          eventValue: eventValue,
        };
        return Object.keys(expectedFields).every(
          (key) => requestBody?.[key] === expectedFields[key]
        );
      }
      return false;
    });
  }
```

Reply:
taken in consideration!